packages/analytics-browser/src/utils/snippet-helper.ts

```diff
@@ -13,10 +14,13 @@ export const runQueuedFunctions = (instance: object, queue: QueueProxy) => {
  */
 export const convertProxyObjectToRealObject = <T>(instance: T, queue: QueueProxy): T => {
   for (let i = 0; i < queue.length; i++) {
-    const [functionName, ...args] = queue[i];
-    const fn = instance && instance[functionName as keyof T];
+    const { name, args, resolve } = queue[i];
+    const fn = instance && instance[name as keyof T];
     if (typeof fn === 'function') {
-      fn.apply(instance, args);
+      const result = fn.apply(instance, args) as AmplitudeReturn<Result>;
+      if (typeof resolve === 'function') {
```

Comment:
looks like this should solve the error below

```suggestion
      if (typeof resolve === 'function' && result) {
```

Reply:
You are right. This solves the error.

Reply:
Only concern here is some promises will remain pending, and never gets resolved

Reply:
Let me update it to resolve the pending issue.